include/ocpp/v201/charge_point.hpp

```diff
@@ -774,6 +778,12 @@ class ChargePoint : public ChargePointInterface, private ocpp::ChargingStationBa
     /// If \param persist is set to true, the change will be persisted across a reboot
     void execute_change_availability_request(ChangeAvailabilityRequest request, bool persist);
 
+protected:
+    std::shared_ptr<SmartChargingHandlerInterface> smart_charging_handler;
```

Comment:
I've mentioned this before but I don't see a good way out of it yet. I don't like making stuff protected just for unit testing. I think we might have to look at a better solution for the library in general to improve testing capabilities. Don't expect you to make any changes based on this comment, if you guys have idea's that's always welcome of course ;)

Reply:
We agree and once we have things more fully delivered we can take a look at improving on this.

Reply:
Last Monday I had a discussion with the people from Pionix. We have some ideas on how to improve the way we can test charge_point and more specifically the message handling. Nobody will pick it up yet due to other priorities but that will help for this issue as well.

Reply:
Yes of course. 

The general idea is to split up the handling of messages per functional block. In this case that would be a class to handle K related messages. The charge point will most likely still receive the messages and forward them to the correct handler. All the dependencies that the handlers might need will be injected by the charge point class. 

With this in place it would be possible to test the handling of the K messages by only constructing an object of the handler and all the other dependencies can be mocked. The charge point does not need to be constructed.

The other way around is also possible. You can construct a charge point with mocked message handlers so you are able to test for example the message routing.

How this will work in practice and if this is the final way of doing it is not sure yet. This was just what we discussed briefly.

Reply:
Would it be useful to add this to an issue within the `EVerest/libocpp` repo so the community can better track this? I do not think we are expected to take any action within this PR.

Reply:
Just to clarify, I am not asking for any changes other than:
- the process for tracking the comments and ensuring that they are addressed
- at least a high-level design of what the final error handling during load will be; even if the implementation waits until later